ospfd/ospf_te.c

```diff
@@ -2557,32 +2557,27 @@ DEFUN (show_ip_ospf_mpls_te_link,
 			return CMD_SUCCESS;
 		}
 		ospf = ospf_lookup_by_inst_name(inst, vrf_name);
-		if (ospf == NULL || !ospf->oi_running)
-			return CMD_SUCCESS;
-		vrf = vrf_lookup_by_id(ospf->vrf_id);
-		FOR_ALL_INTERFACES (vrf, ifp)
-			show_mpls_te_link_sub(vty, ifp);
+	} else
+		ospf = ospf_lookup_by_vrf_id(VRF_DEFAULT);
+	if (ospf == NULL || !ospf->oi_running)
 		return CMD_SUCCESS;
-	}
-	/* Show All Interfaces. */
-	if (argc == 5) {
-		for (ALL_LIST_ELEMENTS_RO(om->ospf, node, ospf)) {
-			if (!ospf->oi_running)
-				continue;
-			vrf = vrf_lookup_by_id(ospf->vrf_id);
-			FOR_ALL_INTERFACES (vrf, ifp)
-				show_mpls_te_link_sub(vty, ifp);
-		}
-	}
-	/* Interface name is specified. */
-	else {
-		ifp = if_lookup_by_name_all_vrf(argv[idx_interface]->arg);
+	vrf = vrf_lookup_by_id(ospf->vrf_id);
+	if (!vrf)
+		return CMD_SUCCESS;
+	if (idx_interface) {
+		ifp = if_lookup_by_name(
+					argv[idx_interface]->arg,
+					ospf->vrf_id);
 		if (ifp == NULL)
-			vty_out(vty, "No such interface name\n");
-		else
+			vty_out(vty, "No such interface name in vrf %s\n",
+				vrf->name);
+	}
+	if (!ifp) {
```

Comment:
If I'm reading the code right, if you enter `show ip ospf mpls-te interface eth500` and `eth500` is not found, the CLI will print an error message ("No such interface name ...") but will proceed to show all interfaces. Please fix this.